### PR TITLE
blockdev_mirror: support ceph backend

### PR DESCRIPTION
**Pull request by aliang123**
Add ceph backend for block mirror

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2046047

### CODE REVIEW

**Comment by aliang123** [comment]
01/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_mirror_cancel_running_job.q35: PASS (61.51 s)
 (02/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_mirror_no_space.q35: PASS (48.85 s)
 (03/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.with_data_plane.q35: PASS (92.77 s)
 (04/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.q35: PASS (101.82 s)
 (05/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512.with_data_plane.q35: PASS (101.51 s)
 (06/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512.q35: PASS (101.68 s)
 (07/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_2M.with_data_plane.q35: PASS (99.83 s)
 (08/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_2M.q35: PASS (101.02 s)
 (09/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_random_cluster_size.with_data_plane.q35: PASS (99.61 s)
 (10/40) Host_RHEL.m8.u6.product_rhel.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_random_cluster_size.q35: PASS (99.32 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny @MiriamDeng, please help to review, thanks.

File: qemu/tests/cfg/blockdev_mirror_error.cfg [hunk 1]
```diff
@@ -40,6 +40,9 @@
         image_create_support_data1 = no
     iscsi_direct:
         lun_data1 = 1
+    ceph:
+        enable_ceph_mirror1 = no
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_mirror_error.cfg:44]: remove this line for we've defined it at line58

File: qemu/tests/cfg/blockdev_mirror_qemuio_ready_job.cfg [hunk 2]
```diff
@@ -26,3 +26,5 @@
     storage_type_default = directory
     qemuio_cmd = '%s -f %s %s'
     error_msg = Failed to get "write" lock
+    ceph:
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_mirror_qemuio_ready_job.cfg:29]: filesystem only for this case, please make a double check if we want to extend the storage backends

File: qemu/tests/cfg/blockdev_mirror_to_rbd.cfg [hunk 3]
```diff
@@ -1,6 +1,5 @@
 - blockdev_mirror_to_rbd:
     only Linux
-    only ceph
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_mirror_to_rbd.cfg:3]: only ceph filesystem

File: qemu/tests/cfg/blockdev_mirror_to_rbd.cfg [hunk 4]
```diff
@@ -30,3 +29,7 @@
     enable_ceph_mirror1 = yes
     enable_ceph_data1 = no
     enable_ceph_image1 = no
+    ceph:
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_mirror_to_rbd.cfg:32]: Please run this case with filesystem and ceph backends, check the logs to see if the qmp commands are expected.

File: qemu/tests/cfg/blockdev_mirror_to_rbd.cfg (hunk 3 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_mirror_to_rbd.cfg]: Have updated it to: only ceph

File: qemu/tests/cfg/blockdev_mirror_qemuio_ready_job.cfg (hunk 2 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_mirror_qemuio_ready_job.cfg]: Have removed it, thanks.

Comment (aliang123) [reply qemu/tests/cfg/blockdev_mirror_qemuio_ready_job.cfg]: Have updated it, thanks zhenchao.

File: qemu/tests/cfg/blockdev_mirror_error.cfg (hunk 1 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_mirror_error.cfg]: Have removed it, thanks zhenchao.

**Comment by MiriamDeng** [comment]
Applied the patch and ran it on ppc, all passed. LGTM
 (01/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.with_data_plane: STARTED
| (01/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.with_data_plane: PASS (166.70 s)
 (02/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test: STARTED
- (02/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test: PASS (221.55 s)
 (03/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512.with_data_plane: STARTED
/ (03/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512.with_data_plane: PASS (206.77 s)
 (04/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512: STARTED
- (04/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512: FAIL: AssertionError("Get file ('data') MD5 with error: md5sum: /mnt/sdb1/data: No such file or directory\n",) (208.78 s)
 (05/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_2M.with_data_plane: STARTED
| (05/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_2M.with_data_plane: PASS (161.29 s)
 (06/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_2M: STARTED
- (06/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_2M: PASS (157.84 s)
 (07/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_random_cluster_size.with_data_plane: STARTED
| (07/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_random_cluster_size.with_data_plane: PASS (165.79 s)
 (08/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_random_cluster_size: STARTED
\ (08/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_random_cluster_size: PASS (247.46 s)
 (09/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_no_space: STARTED
/ (09/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_no_space: PASS (81.17 s)
 (10/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_cancel_running_job: STARTED
| (10/10) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_cancel_running_job: PASS (98.05 s

(1/2) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512.with_data_plane: STARTED
\ (1/2) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512.with_data_plane: PASS (209.51 s)
 (2/2) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512: STARTED
| (2/2) Host_RHEL.m8.u6.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_full_mirror.simple_test.dst_cluster_size_512: PASS (252.71 s)

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**